File: app/src/main/java/com/dgc/androidframe/test_activity/TestActivity.java
package com.dgc.androidframe.test_activity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.dgc.androidframe.R;
import com.dgc.androidframe.base.BaseActivity;


public class TestActivity extends BaseActivity implements TestView {
    TextView text;
    TestPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);
        text = (TextView) findViewById(R.id.text);
        //初始化Presenter
        presenter = new TestPresenter();
        presenter.attachView(this);
        addPresenters(presenter);
    }

    public void btnClick(View v) {
        presenter.getData("normal");
    }

    @Override
    public void showData(String data) {
        text.setText(data);
    }

}

File: app/src/main/java/com/dgc/androidframe/test_activity/TestActivity.kt
package com.dgc.androidframe.test_activity

import android.os.Bundle
import android.view.View
import android.widget.TextView

import com.dgc.androidframe.R
import com.dgc.framework.base.BaseActivity

class TestActivity : BaseActivity(), TestView {
    private lateinit var text: TextView
    private lateinit var presenter: TestPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_test)
        text = findViewById(R.id.text)

        presenter = TestPresenter()
        presenter.attachView(this)
        addPresenters(presenter)
    }

    fun btnClick(v: View?) {
        presenter.getData("normal")
    }

    override fun showData(data: String?) {
        text.text = data
    }
}